pkg/devfile/parser/data/interface.go

```diff
@@ -26,26 +26,30 @@ type DevfileData interface {
 	GetComponents(common.DevfileOptions) ([]v1.Component, error)
 	AddComponents(components []v1.Component) error
 	UpdateComponent(component v1.Component)
+	DeleteComponent(name string) error
 
 	// project related methods
 	GetProjects(common.DevfileOptions) ([]v1.Project, error)
 	AddProjects(projects []v1.Project) error
 	UpdateProject(project v1.Project)
+	DeleteProject(name string) error
 
 	// starter projects related commands
 	GetStarterProjects(common.DevfileOptions) ([]v1.StarterProject, error)
 	AddStarterProjects(projects []v1.StarterProject) error
 	UpdateStarterProject(project v1.StarterProject)
+	DeleteStarterProject(name string) error
 
 	// command related methods
 	GetCommands(common.DevfileOptions) ([]v1.Command, error)
-	AddCommands(commands ...v1.Command) error
+	AddCommands(commands []v1.Command) error
```

Comment:
This is changing the signature of an existing function. It will break every project that already uses this function.

Reply:
@kadel Thanks for the review and concern. Yes, you're right, there should be a release every sprint, specifically when there is a breaking change with the changelog of the PR/commit.

Let me think about it a bit more. Since devfile/library has never had a release, I can do a v1.0.0-alpha release after this PR.

What do you think?

Reply:
That would be great. At least we would know when and what problems we can expect when upgrading to a newer version.

Reply:
@kadel do you have any other concerns before I merge this PR? I am looking to do a pre-release once this PR is in and hopefully generate a changelog for the release so its easier to track changes.